tracer/src/Datadog.Trace/Iast/Aspects/System.Text.Json/JsonDocumentAspects.cs

```diff
@@ -0,0 +1,126 @@
+// <copyright file="JsonDocumentAspects.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+#if NETCOREAPP3_0_OR_GREATER
+using System;
+using System.Text.Json;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Iast.Dataflow;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Iast.Aspects.System.Text.Json;
+
+/// <summary> System.Text.Json JsonDocument class aspect </summary>
+[AspectClass("System.Text.Json", AspectType.Source)]
+[global::System.ComponentModel.Browsable(false)]
+[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+public class JsonDocumentAspects
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<JsonDocumentAspects>();
+
+    /// <summary>
+    /// Parse method aspect
+    /// Taint all Parent from JsonElement that are string
+    /// </summary>
+    /// <param name="json">the JsonDocument result of Parse</param>
+    /// <param name="options">the JsonDocumentOptions</param>
+    /// <returns>the JsonDocument result</returns>
+    [AspectMethodReplace("System.Text.Json.JsonDocument::Parse(System.String,System.Text.Json.JsonDocumentOptions)")]
+    public static object Parse(string json, JsonDocumentOptions options)
+    {
+        var doc = JsonDocument.Parse(json, options);
+
+        try
+        {
+            TaintJsonElements(json, doc);
+        }
+        catch (Exception ex)
+        {
+            Log.Warning(ex, "Error tainting JsonDocument.Parse result");
+        }
+
+        return doc;
+    }
+
+    /// <summary>
+    /// GetString and GetRawText method aspect
+    /// Taint the string result when the parent is tainted
+    /// </summary>
+    /// <param name="target">the JsonElement instance</param>
+    /// <returns>the string result</returns>
+    [AspectMethodReplace("System.Text.Json.JsonElement::GetString()", [0], [true])]
+    [AspectMethodReplace("System.Text.Json.JsonElement::GetRawText()", [0], [true])]
+    public static string? GetString(object target)
+    {
+        IJsonElement? element;
+        try
+        {
+            element = target.DuckCast<IJsonElement>();
```

Comment:
If you're referencing System.Text.Json (you are) then do you need to use duck typing at all? 🤔

Reply:
I'm using duck typing here to get the value of the private object `_parent`